android/src/main/java/sqip/flutter/SquareInAppPaymentsFlutterPlugin.java

```diff
@@ -28,140 +30,121 @@
-    // So there will be null pointer exception like this issue: https://github.com/square/in-app-payments-flutter-plugin/issues/150
-    // We move the following code to onAttachedToActivity and onReattachedToActivityForConfigChanges, which will be only called once,
-    // in order to avoid creating an invalid CardEntryModule and GooglePayModule.
-    // channel.setMethodCallHandler(this);
-    // cardEntryModule = new CardEntryModule(channel);
-    // googlePayModule = new GooglePayModule(channel);
+  public void onAttachedToEngine(FlutterPluginBinding binding) {
+    this.applicationContext = binding.getApplicationContext();
+    channel = new MethodChannel(binding.getBinaryMessenger(), "square_in_app_payments");
+    // MethodCallHandler is set in onAttachedToActivity to avoid NPE issues
   }
 
   @Override
-  public void onDetachedFromEngine(FlutterPluginBinding flutterPluginBinding) {
+  public void onDetachedFromEngine(FlutterPluginBinding binding) {
     cardEntryModule = null;
     googlePayModule = null;
     channel = null;
   }
 
   @Override
-  public void onAttachedToActivity(ActivityPluginBinding activityPluginBinding) {
-    // KNOWN ISSUE: OnAttachedToEngine can be called twice which may be due to https://github.com/flutter/flutter/issues/69721
-    // Once the issue is resolved, we can check if we can move following three lines to OnAttachedToEngine.
+  public void onAttachedToActivity(ActivityPluginBinding activityBinding) {
     channel.setMethodCallHandler(this);
-    cardEntryModule = new CardEntryModule(channel);
-    googlePayModule = new GooglePayModule(channel);
-    googlePayModule.attachActivityResultListener(activityPluginBinding, channel);
-    cardEntryModule.attachActivityResultListener(activityPluginBinding, channel);
+    cardEntryModule = new CardEntryModule(applicationContext, channel);
+    googlePayModule = new GooglePayModule(applicationContext, channel);
+
+    googlePayModule.attachActivityResultListener(activityBinding, channel);
+    cardEntryModule.attachActivityResultListener(activityBinding, channel);
   }
 
   @Override
-  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding activityPluginBinding) {
-    // KNOWN ISSUE: OnAttachedToEngine can be called twice which may be due to https://github.com/flutter/flutter/issues/69721
-    // Once the issue is resolved, we can check if we can move following three lines to OnAttachedToEngine.
-    channel.setMethodCallHandler(this);
-    cardEntryModule = new CardEntryModule(channel);
-    googlePayModule = new GooglePayModule(channel);
-    googlePayModule.attachActivityResultListener(activityPluginBinding, channel);
-    cardEntryModule.attachActivityResultListener(activityPluginBinding, channel);
+  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding activityBinding) {
+    onAttachedToActivity(activityBinding);
```

Comment:
It's probably light enough not to matter, but there is a reason a config change breaks the two modules here?  That is, could we not discard them in _all three_ of `detachedFromActivity`, `detachedFromEngine`, _and_ `detachedFromActivityForConfigChanges` and re-create what I assume is the same thing all those times?

At the least, I bet you never detach from engine except after detaching from activity, so you could instead `check(cardEntryModule == null)` in `onDetachFromEngine`, because it should already have been done?